src/main/scala/li/cil/oc/integration/appeng/DriverExportBus.scala

```diff
@@ -12,86 +21,90 @@ import li.cil.oc.util.ExtendedArguments._
-                case _: NoSuchMethodException => export.getConfigManager.getClass.getMethod("getSetting", classOf[Settings])
-              }).invoke(export.getConfigManager, Settings.FUZZY_MODE).asInstanceOf[FuzzyMode]
-              val source = new MachineSource(export)
-              var didSomething = false
-              for (slot <- 0 until config.getSizeInventory if count > 0) {
-                val filter = AEApi.instance.storage.createItemStack(config.getStackInSlot(slot))
-                val stacks =
-                  if (export.getInstalledUpgrades(Upgrades.FUZZY) > 0)
-                    itemStorage.getStorageList.findFuzzy(filter, fuzzyMode).toSeq
-                  else
-                    Seq(itemStorage.getStorageList.findPrecise(filter))
-                for (ais <- stacks.filter(_ != null).map(_.copy()) if count > 0) {
-                  val is = ais.getItemStack
-                  is.stackSize = count
-                  if (InventoryUtils.insertIntoInventorySlot(is, inventory, Option(side.getOpposite), targetSlot, count, simulate = true)) {
-                    ais.setStackSize(count - is.stackSize)
-                    val eais = AEApi.instance.storage.poweredExtraction(export.getProxy.getEnergy, itemStorage, ais, source)
-                    if (eais != null) {
-                      val eis = eais.getItemStack
-                      count -= eis.stackSize
-                      didSomething = true
-                      InventoryUtils.insertIntoInventorySlot(eis, inventory, Option(side.getOpposite), targetSlot)
-                      if (eis.stackSize > 0) {
-                        eais.setStackSize(eis.stackSize)
-                        itemStorage.injectItems(ais, Actionable.MODULATE, source)
-                      }
+      val part = host.getPart(side)
+
+      if(AEUtil.isExportBus(part.getItemStack(PartItemStack.PICK))) {
+        val export = part.asInstanceOf[IGridHost with ISegmentedInventory with IConfigurableObject with IUpgradeableHost with IActionHost]
+        InventoryUtils.inventoryAt(new BlockPosition(host.getLocation.x, host.getLocation.y, host.getLocation.z), side) match {
+          case Some(inventory) =>
+            val targetSlot = args.checkSlot(inventory, 1)
+            val config = export.getInventoryByName("config")
+            val itemStorage = AEUtil.getGridStorage(export.getGridNode(AEPartLocation.fromFacing(side)).getGrid).getItemInventory
+            var count = export.getInstalledUpgrades(Upgrades.SPEED) match {
+              case 1 => 8
+              case 2 => 32
+              case 3 => 64
+              case 4 => 96
+              case _ => 1
+            }
+            // We need reflection here to avoid compiling against the return and
+            // argument type, which has changed in rv2-beta-20 or so.
+            val fuzzyMode = (try export.getConfigManager.getClass.getMethod("getSetting", classOf[Enum[_]]) catch {
```

Comment:
Sorry but i don't fully understand the fuzzy-mode code and i will not remove this if i don't exactly know what this means